open_spiel/games/phantom_go/phantom_go_board.cc

```diff
@@ -0,0 +1,868 @@
+  }
+
+  last_ko_point_ = kInvalidPoint;
+}
+
+bool GoBoard::PlayMove(VirtualPoint p, GoColor c) {
+  if (p == kVirtualPass) {
+    last_ko_point_ = kInvalidPoint;
+    return true;
+  }
+
+  /*int boardPoint = VirtualPointToBoardPoint(p, board_size_);
+
+  printf("playing boardPoint %i, check %i\n", boardPoint, VirtualPointFromBoardPoint(boardPoint, board_size_));*/
+
+  //std::vector<GoColor> currObservation;
+  
+  //update 1
+  //add observation to current player's observation
+
+  if (c == GoColor::kBlack)
+  {
+      observationBlack[VirtualPointToBoardPoint(p, board_size_)] = board_[p].color;
+  }
+  else
+  {
+      observationWhite[VirtualPointToBoardPoint(p, board_size_)] = board_[p].color;
+  }
+
+  //currObservation[p] = board_[p].color;
+
+  /*if (board_[p].color != GoColor::kEmpty) {
+    SpielFatalError(absl::StrCat("Trying to play the move ", GoColorToString(c),
+                                 ": ", VirtualPointToString(p), " (", p,
+                                 ") but the cell is already filled with ",
+                                 GoColorToString(board_[p].color)));
+  }
+  SPIEL_CHECK_EQ(GoColor::kEmpty, board_[p].color);*/
+
+  //update 1
+  //playing illegal moves will occur standardly during phantom go, it is even desired  
+  if (IsLegalMoveObserver(p, c) == false)
+  {
+      return false;
+  }
+
+  if (c == GoColor::kBlack)
+  {
+      stoneCount.second++;
+  }
+  else
+  {
+      stoneCount.first++;
+  }
+
+  // Preparation for ko checking.
+  bool played_in_enemy_eye = true;
+  Neighbours(p, [this, c, &played_in_enemy_eye](VirtualPoint n) {
+    GoColor s = PointColor(n);
+    if (s == c || s == GoColor::kEmpty) {
+      played_in_enemy_eye = false;
+    }
+  });
+
+  JoinChainsAround(p, c);
+  SetStone(p, c);
+  RemoveLibertyFromNeighbouringChains(p);
+  int stones_captured = CaptureDeadChains(p, c);
+
+  if (stones_captured)
+  {
+      if (c == GoColor::kBlack)
+      {
+          stoneCount.first -= stones_captured;
+      }
+      else
+      {
+          stoneCount.second-= stones_captured;
+      }
+  }
+
+  //update 5
+  //add own stone to own observation
+
+  if (c == GoColor::kBlack)
+  {
+      observationBlack[VirtualPointToBoardPoint(p, board_size_)] = GoColor::kBlack;
+  }
+  else
+  {
+      observationWhite[VirtualPointToBoardPoint(p, board_size_)] = GoColor::kWhite;
+  }
+
+  
+
+  if (played_in_enemy_eye && stones_captured == 1) {
+    last_ko_point_ = last_captures_[0];
+  } else {
+    last_ko_point_ = kInvalidPoint;
+  }
+
+  //update 2
+  //if player captured stones, update his observation
+
+  if (stones_captured != 0)
+  {
+      printf("removing points\n");
+      for (int point = 0; point < board_size_ * board_size_; point++)
+      {
+          
+          VirtualPoint vpoint = VirtualPointFromBoardPoint(point, board_size_);
+
+          // example: if current color is white, compare observation of black, where all black stones are in state before removal, to observer board
```

Comment:
Use proper English sentences, just as if you wrote a document (capital letters, punctuation, no shortcuts).

Reply:
I am going to remove this comment, I have written it for myself to more easily write the following code.